lib/postgrex/protocol.ex

```diff
@@ -2093,7 +2105,7 @@ defmodule Postgrex.Protocol do
         bind_execute(s, status, query, params)
 
       {error, _, _} = other when error in [:error, :disconnect] ->
-        other
+        handle_disconnect_retry(other)
```

Comment:
This part looks like what happens if we try to execute a prepared statement from the cache but it's not on the connection we were given from the pool. So it is basically an error during the prepare step